src/components/Table/HeadingCell.react.js

```diff
@@ -3,11 +3,47 @@ export class HeadingCell extends React.Component {
     super(props);
   }
 
+  queryName() {
+    return "order[" + this.props.sortable + "]";
+  }
+
+  currentSortOrder() {
+    return this.context.tableStore.getUrl().query(true)[this.queryName()];
+  }
+
+  nextSortOrder() {
+    return this.currentSortOrder() === "asc" ? "desc" : "asc";
```

Comment:
Possibly, though `currentSortOrder` will predominantly be `undefined`—`reversedSortOrder` wouldn't capture the intent (as we rarely would have a *sort order* to reverse).